src/test/javascript/spec/component/settings/ide/ide-settings.component.spec.ts

```diff
@@ -0,0 +1,112 @@
+import { ComponentFixture, TestBed } from '@angular/core/testing';
+import { CUSTOM_ELEMENTS_SCHEMA } from '@angular/core';
+import { of } from 'rxjs';
+import { IdeSettingsComponent } from 'app/shared/user-settings/ide-preferences/ide-settings.component';
+import { IdeSettingsService } from 'app/shared/user-settings/ide-preferences/ide-settings.service';
+import { ProgrammingLanguage } from 'app/entities/programming-exercise.model';
+import { HttpClientTestingModule } from '@angular/common/http/testing';
+
+describe('IdeSettingsComponent', () => {
+    let component: IdeSettingsComponent;
+    let fixture: ComponentFixture<IdeSettingsComponent>;
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    let ideSettingsService: IdeSettingsService;
+
+    const mockIdeSettingsService = {
+        loadPredefinedIdes: jest.fn(),
+        loadIdePreferences: jest.fn(),
+        saveIdePreference: jest.fn(),
+        deleteIdePreference: jest.fn(),
+    };
+
+    beforeEach(async () => {
```

Comment:
To avoid the tests interfering with each other, I'd recommend adding an afterEach hook as well.
```
    afterEach(() => {
        jest.resetAllMocks();
    });
```

Then, you could be more specific with your assertions and replace 

- `toHaveBeenCalled` with `toHaveBeenCalledOnce` and 
- `toHaveBeenCalledWith` with `toHaveBeenCalledExactlyOnceWith`.

Reply:
done